### PR TITLE
[GNA] Add extra segments to PWL only if it is fused with Conv2D layer

### PR DESCRIPTION
**Pull request by marcinkusm**
### Details:
 - Added checking if PWL layers is fused with Conv2D layer

### Tickets:
 - 54920

### CODE REVIEW

File: src/plugins/intel_gna/runtime/pwl.cpp [hunk 1]
```diff
@@ -39,36 +39,80 @@ inline double power(const double x, const std::tuple<double, double, double>& ar
...
         case kActRelu:
-            make_gna_pwl(activation_type, pwl, -1.0, 1.0, scale_in, scale_out, low_precision, ptr_segment);
+            make_gna_pwl(activation_type,
+                         pwl,
+                         -1.0,
+                         1.0,
+                         scale_in,
+                         scale_out,
+                         low_precision,
+                         is_fused_with_conv2d, ptr_segment);
```
Comment (sirzabek) [inline src/plugins/intel_gna/runtime/pwl.cpp:62]:
```suggestion
                         is_fused_with_conv2d,
                         ptr_segment);
```

File: src/plugins/intel_gna/gna_graph_compiler.cpp [hunk 2]
```diff
@@ -2068,12 +2068,17 @@ case name:\
                 output_pwl_scale_factor,
                 gnaFlags->input_low_precision);
         } else {
+            auto is_fused_with_conv_2d = false;
+            if (LayerInfo(prevLayer).isConvolution()) {
```
Comment (sirzabek) [inline src/plugins/intel_gna/gna_graph_compiler.cpp:2072]: Cannot dnnComponents be used here, checking for kDnnConvolutional2dOp?

**Review by sirzabek — CHANGES_REQUESTED** [review]
Any tests?

File: src/plugins/intel_gna/backend/make_pwl.cpp [hunk 3]
```diff
@@ -322,7 +275,7 @@ void make_gna_pwl(const DnnActivation&  fun,
         default:
             THROW_GNA_EXCEPTION << "Unexpected function activation!" << fun;
     }
-    insert_extra_pwl_segments(gna_pwl, y_min, y_max);
+    insert_extra_pwl_segments(gna_pwl, y_min, y_max, is_fused_with_conv2d);
```
[the remark this answers is not in the document]
Comment (marcinkusm) [reply src/plugins/intel_gna/backend/make_pwl.cpp:278]: I thought to do this in advance to my second PR where this condition will be checked in two places, but I will change it here.

**Review by sirzabek — APPROVED** [review]

**Review by sirzabek — CHANGES_REQUESTED** [review]
Would be nice to have some (unit) tests.

**Comment by marcinkusm** [comment]
If this PR will be merged first unit tests in following PR will be extended: https://github.com/openvinotoolkit/openvino/pull/12825

File: src/plugins/intel_gna/gna_graph_compiler.cpp [hunk 4]
```diff
@@ -2068,12 +2068,19 @@ case name:\
                 output_pwl_scale_factor,
                 gnaFlags->input_low_precision);
         } else {
+            auto is_fused_with_conv_2d = false;
+            auto prevDnnLayer = dnnComponents.findComponent(prevLayer);
+            if (prevDnnLayer && prevDnnLayer->operation == kDnnConvolutional2dOp) {
+                is_fused_with_conv_2d = true;
+            }
+
             PwlDesignOpt(activation_type,
```
Comment (kbruniec) [inline src/plugins/intel_gna/gna_graph_compiler.cpp:2077]: Shouldn't we cover `PwlDesign` too (uniformPwlDesign)?

Comment (marcinkusm) [reply src/plugins/intel_gna/gna_graph_compiler.cpp:2077]: Inserting of extra segments is currently added only to the PwlDesignOpt().
Should I add this also to the PwlDesign()?

File: src/plugins/intel_gna/gna_graph_compiler.cpp [hunk 5]
```diff
@@ -2068,12 +2068,19 @@ case name:\
                 output_pwl_scale_factor,
                 gnaFlags->input_low_precision);
         } else {
+            auto is_fused_with_conv_2d = false;
+            auto prevDnnLayer = dnnComponents.findComponent(prevLayer);
+            if (prevDnnLayer && prevDnnLayer->operation == kDnnConvolutional2dOp) {
```
[the remark this answers is not in the document]
Comment (marcinkusm) [reply src/plugins/intel_gna/gna_graph_compiler.cpp:2073]: I am not sure, previously it was done by checking previous CNNLayer and verifying if it is convolution2D, but I understand that in both cases we will have the same situation.

[the remark this answers is not in the document]
Comment (marcinkusm) [reply src/plugins/intel_gna/gna_graph_compiler.cpp]: It is fixed and checking of pooling is reflected

File: src/plugins/intel_gna/gna_graph_compiler.cpp (hunk 4 above)
Comment (marcinkusm) [reply src/plugins/intel_gna/gna_graph_compiler.cpp:2090]: It was discussed with @nkogteva and @dorloff and decision is that adding limitation for extra segments of errata will be updated only for the place in code when errata was applied until now.

uniformPwlDesign option is not clear and is suppose to be cleared as deprecated. There will be created separate cleanup ticket created  for that by @dorloff.

File: src/plugins/intel_gna/gna_graph_compiler.cpp (hunk 5 above)
[the remark this answers is not in the document]
Comment (kbruniec) [reply src/plugins/intel_gna/gna_graph_compiler.cpp]: great!

File: src/plugins/intel_gna/gna_graph_compiler.cpp (hunk 4 above)
Comment (kbruniec) [reply src/plugins/intel_gna/gna_graph_compiler.cpp:2090]: I am fine with that.

**Review by kbruniec — APPROVED** [review]
LGTM

**Review by sirzabek — APPROVED** [review]

File: src/tests/unit/gna/gna_extra_pwl_segments_tests.cpp [hunk 6]
```diff
@@ -0,0 +1,239 @@
+// Copyright (C) 2018-2022 Intel Corporation
```
Comment (dorloff) [inline src/tests/unit/gna/gna_extra_pwl_segments_tests.cpp:1]: If it's a new file, it should be just "2022"

**Pull request merged**